### PR TITLE
add flag for block period

### PR DESCRIPTION
**Pull request by mikedoan**
add flag for block period

### CODE REVIEW

File: cmd/harmony/main.go [hunk 1]
```diff
@@ -101,8 +100,11 @@ var (
 	isExplorer = flag.Bool("is_explorer", false, "true means this node is a node to serve explorer")
 	// networkType indicates the type of the network
 	networkType = flag.String("network_type", "mainnet", "type of the network: mainnet, testnet, devnet...")
+	// blockPeriod indicates the how long the leader waits to propose a new block.
+	blockPeriod = flag.Int("block_period", 10, "how long in second the leader waits to propose a new block.")
```
Comment (rlan35) [inline cmd/harmony/main.go]: This still won't gaurantee exact 10s block interval. It only adds 10s delay after the consensus is reached. If consensus take longer, the block interval will be longer. The best solution is to have a timer for 10s starting from consensus start and propose the next block when the 10s timeout is reached and consensus was reached before 10s. If consensus takes more than 10s, then propose next block immediately thereafter.

Comment (LeoHChen) [reply cmd/harmony/main.go]: agree, that's the best one.  @mikedoan , can you add another change to support that?

File: node/node_newblock.go [hunk 2]
```diff
@@ -54,7 +56,6 @@ func (node *Node) WaitForConsensusReadyv2(readySignal chan struct{}, stopChan ch
 				}
 			case <-readySignal:
 				firstTry := true
-				deadline := time.Now().Add(BlockPeriod)
 				for {
 					if !firstTry {
 						time.Sleep(PeriodicBlock)
```
Comment (LeoHChen) [inline node/node_newblock.go]: what's the use of this sleep?

**Comment by LeoHChen** [comment]
Also, please test locally to make sure the 10s interval is right.

**Comment by mikedoan** [comment]
```
DEBUG[06-21|16:35:11.686] [ConsensusMainLoop] STARTING CONSENSUS   port=9000 ip=127.0.0.1 myBlock=1 myViewID=0 phase=Announce mode=Normal numTxs=2 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:35:11-0700 publicKeys=20
INFO [06-21|16:35:13.997] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000 ip=127.0.0.1 myBlock=2 myViewID=1 phase=Announce mode=Normal BlockNum=1 ViewId=0 BlockHash=0372d1…959ce2                                                                                                         index=0
DEBUG[06-21|16:35:22.033] [ConsensusMainLoop] STARTING CONSENSUS   port=9000  ip=127.0.0.1 myBlock=2 myViewID=1 phase=Announce mode=Normal numTxs=0 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:35:22-0700 publicKeys=20
INFO [06-21|16:35:22.512] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000  ip=127.0.0.1 myBlock=3 myViewID=2 phase=Announce mode=Normal BlockNum=2 ViewId=1 BlockHash=7b5f9f…c7518e                                                                                                           index=0
DEBUG[06-21|16:35:32.541] [ConsensusMainLoop] STARTING CONSENSUS   port=9000  ip=127.0.0.1 myBlock=3 myViewID=2 phase=Announce mode=Normal numTxs=0 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:35:32-0700 publicKeys=20
INFO [06-21|16:35:32.921] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000  ip=127.0.0.1 myBlock=4 myViewID=3 phase=Announce mode=Normal BlockNum=3 ViewId=2 BlockHash=b2ec9b…ca3f3f                                                                                                           index=0
DEBUG[06-21|16:35:42.961] [ConsensusMainLoop] STARTING CONSENSUS   port=9000  ip=127.0.0.1 myBlock=4 myViewID=3 phase=Announce mode=Normal numTxs=0 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:35:42-0700 publicKeys=20
INFO [06-21|16:35:43.424] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000  ip=127.0.0.1 myBlock=5 myViewID=4 phase=Announce mode=Normal BlockNum=4 ViewId=3 BlockHash=0d98b2…206af7                                                                                                           index=0
DEBUG[06-21|16:35:53.470] [ConsensusMainLoop] STARTING CONSENSUS   port=9000  ip=127.0.0.1 myBlock=5 myViewID=4 phase=Announce mode=Normal numTxs=0 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:35:53-0700 publicKeys=20
INFO [06-21|16:35:53.830] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000  ip=127.0.0.1 myBlock=6 myViewID=5 phase=Announce mode=Normal BlockNum=5 ViewId=4 BlockHash=550d3e…55c611                                                                                                           index=0
DEBUG[06-21|16:36:03.942] [ConsensusMainLoop] STARTING CONSENSUS   port=9000  ip=127.0.0.1 myBlock=6 myViewID=5 phase=Announce mode=Normal numTxs=0 consensus="[duty:LDR, PubKey:ca23704be46ce9c4704681ac9c08ddc644f1858a5c28ce236e1b5d9dee67c1f5a28075b5ef089adeffa8a372c1762007, ShardID:0]" startTime=2019-06-21T16:36:03-0700 publicKeys=20
INFO [06-21|16:36:04.414] HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!   port=9000  ip=127.0.0.1 myBlock=7 myViewID=6 phase=Announce mode=Normal BlockNum=6 ViewId=5 BlockHash=1af81b…89f143                                                                                                            index=0
```

ran in local - and saw consensus every 10s

**Pull request merged**